### PR TITLE
output direction change for mech fabs

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
this adds the ability to change the output direction (the direction) of the mech fabricator using alt-click whilst it has its wire panel open.

## Why It's Good For The Game
Certain maps have a dir for the mech fabricator that prints in that dir; however, if it were to be destroyed, you can't change its dir whatsoever. 

## Changelog

:cl:
qol: You can now rotate the direction of the mecha fab if its wire panel is open
/:cl:

### CODE REVIEW

**Comment by jjpark-kb** [comment]
Please tell me if this isn't a qol and I will change it... I am unsure.

**Comment by MrMelbert** [comment]
Any reason in particular why this doesn't use `/datum/component/simple_rotation`?

**Comment by jjpark-kb** [comment]
> Any reason in particular why this doesn't use `/datum/component/simple_rotation`?

Ah... that's a thing? That would be the reason, I just didn't know about that. Let me do that real quick.

**Comment by jjpark-kb** [comment]
I'll continue working on it... I am just having a difficult time for no good reason adding this component.

**Comment by jjpark-kb** [comment]
I am unsure of why the simple_rotation component is not working. I really wanted it to be where it requires the panel open, but then realized that the can_interact proc prevents it from working because it checks if the panel is open and if it doesn't have a specific flag; however, the INTERACT_MACHINE_OPEN flag would then also allow it to open the menu of the mecha fabricator whilst its panel is open, which would not be intended. 

I think I'm going to stick to the current code... I apologize.

File: code/modules/vehicles/mecha/mech_fabricator.dm [hunk 1]
```diff
@@ -112,6 +112,17 @@
...
+	if(panel_open)
+		. += span_notice("Alt-click to rotate the output direction.")
+
+/obj/machinery/mecha_part_fabricator/AltClick(mob/user)
+	. = ..()
+	if(!user.canUseTopic(src, BE_CLOSE))
+		return
+	if(panel_open)
+		dir = turn(dir, -90)
+		to_chat(user, span_notice("You change [src]'s output direction to [dir2text(dir)]."))
```
Comment (Mothblocks) [inline code/modules/vehicles/mecha/mech_fabricator.dm:124]:
```suggestion
		balloon_alert(user, "changed to [dir2text(dir)]"))
```

**Comment by Ryll-Ryll** [comment]
ehh it's fine

**Pull request merged**